tests/plugins/widget/integration/selection/inputselection.js

```diff
@@ -0,0 +1,123 @@
+/* bender-tags: widgetcore */
+/* bender-ckeditor-plugins: widget,undo */
+/* bender-ckeditor-remove-plugins: tableselection */
+
+( function() {
+	'use strict';
+
+	bender.editor = {
+		config: {
+			allowedContent: true,
+			on: {
+				instanceReady: function( evt ) {
+					var editor = evt.editor;
+
+					editor.widgets.add( 'testwidget', {
+						requiredContent: 'div(testwidget)',
+						allowedContent: 'div(testwidget)',
+						template: '<div class="testwidget">' +
+							'<textarea></textarea>' +
+							'</div>',
+						button: 'Add textarea widget',
+						upcast: function( element ) {
+							return element.name === 'div' && element.hasClass( 'testwidget' );
+						},
+						init: function() {
+							this.on( 'focus', function() {
+								this.element.findOne( 'textarea' ).focus();
+							} );
+						}
+					} );
+
+					editor.widgets.add( 'testwidget2', {
+						requiredContent: 'div(testwidget2)',
+						allowedContent: 'div(testwidget2)',
+						template: '<div class="testwidget2">' +
+							'<input type="text">' +
+							'</div>',
+						button: 'Add input widget',
+						upcast: function( element ) {
+							return element.name === 'div' && element.hasClass( 'testwidget2' );
+						},
+						init: function() {
+							this.on( 'focus', function() {
+								this.element.findOne( 'input' ).focus();
+							} );
+						}
+					} );
+				}
+			}
+		}
+	};
+
+	bender.test( {
+		'test textarea inside widget should allow on typing': function() {
+			if ( !CKEDITOR.env.ie || CKEDITOR.env.version < 11 ) {
```

Comment:
Looks like the test should be enabled for at least all IE browsers https://github.com/ckeditor/ckeditor4/blob/0c34fecc92e8183d2a15c17237e0fde6730321b3/core/selection.js#L942

on the other hand, AFAIK we don't have any tests confirming that this case will work on other browsers (like Chrome, FF). Maybe it makes sense to ignore browser sniffing to also confirm input elements everywhere?

Reply:
> AFAIK we don't have any tests confirming that this case will work on other browsers (like Chrome, FF). Maybe it makes sense to ignore browser sniffing to also confirm input elements everywhere?

Unfortunately, these tests fail on FF or Chrome. Also, the entire block responsible for the bug was present in the section dedicated to IE:
https://github.com/ckeditor/ckeditor4/blob/cc1939111a2cddfec9019b9105dbde2a84372a1e/core/selection.js#L942-L954

From that reason is really unlikely that bug could appear on non-ie browsers. That's why I feel like having manual tests for other browsers would be sufficient in these cases.
Please also remember that we should immediately know that something is wrong on Chrome or FireFox, because the spreadsheet plugin would fail on those browsers.